src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -398,31 +398,26 @@ private void AddCredentialToCache(Uri serverUri, CURLAUTH authAvail, NetworkCred
             }
         }
 
-        private void AddResponseCookies(Uri serverUri, HttpResponseMessage response)
+        private void AddResponseCookies(EasyRequest state, string cookieHeader)
         {
             if (!_useCookie)
             {
                 return;
             }
 
-            if (response.Headers.Contains(HttpKnownHeaderNames.SetCookie))
+            Uri serverUri = state._requestMessage.RequestUri;
```

Comment:
Sorry I didn't understand this, @kapilash. Didn't we discuss that this is the behavior we need ie. the cookie container should contain the target (new) url? You had mentioned that if we were to use the old url, on the second hop, a cookie would not be found for the new url and the test case would fail